### PR TITLE
New routing event

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/main.rs [hunk 1]
```diff
@@ -25,11 +25,13 @@
 #![deny(deprecated, drop_with_repr_extern, improper_ctypes, missing_docs,
         non_shorthand_field_patterns, overflowing_literals, plugin_as_library,
         private_no_mangle_fns, private_no_mangle_statics, raw_pointer_derive, stable_features,
-        unconditional_recursion, unknown_lints, unsafe_code, unused, unused_allocation,
+        unconditional_recursion, unknown_lints, unsafe_code, unused_allocation,
```
Comment (Fraser999) [inline src/main.rs]: We shouldn't be doing this should we?

Comment (maqi) [inline src/main.rs]: have to do it temporarily as there are much un-implemented code blocks.
mark this done and we shall put it back once integration work is done.

Comment (Fraser999) [inline src/main.rs]: I've already reverted this locally and have the code compiling again (albeit with quite a few `#[allow(dead_code)]` instances).

I think modifying the project lint settings should be last resort, then slightly less worse are file-level overrides, then least worst are function/variable-level overrides.

**Pull request merged**